src/ProxyClient/Cloudflare.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\ProxyClient;
+
+use FOS\HttpCache\ProxyClient\Invalidation\ClearCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\PurgeCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\TagCapable;
+
+/**
+ * Cloudflare HTTP cache invalidator.
+ *
+ * Additional constructor options:
+ * - zone_identifier       Identifier for your Cloudflare zone you want to purge the cache for
+ * - authentication_token  API authorization token, requires Zone.Cache Purge permissions
+ *
+ * @author Simon Jones <simon@studio24.net>
+ */
+class Cloudflare extends HttpProxyClient implements ClearCapable, PurgeCapable, TagCapable
+{
+    /**
+     * @see https://api.cloudflare.com/#getting-started-endpoints
+     */
+    private const API_ENDPOINT = 'https://api.cloudflare.com/client/v4';
```

Comment:
the current logic in the HttpDispatcher (clearly driven by how things work with local caching proxies) is that it is configured with a list of server domain names/ips and sends a copy of the request to each proxy. so e.g. a refresh request for mydomain.com/path would be sent to the host varnish-proxy1.mydomain.com and varnish-proxy2.mydomain.com with a `Host` header of mydomain.com and the /path.

so this domain will be overwritten by the dispatcher. it would be better to have /client/v4 here, without the domain.

alternatively we would need to refactor the HttpDispatcher for this use case, to keep the domain intact. that would make more sense. i think best we would create a base dispatcher that does not have the multiplex logic at all, and have the current HttpDispatcher extend that. then it will be possible to use the base dispatcher for this proxy client instead of the HttpDispatcher that provides unnecessary functionality.

Reply:
>  it would be better to have /client/v4 here, without the domain.

Thanks for confirming, have updated this.